### PR TITLE
UAR-382: Update and move statement pages

### PR DESCRIPTION
**Pull request by joshuahigginsonCH**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-382

### Change description
Moving of statement pages further down the flow

### Work checklist

- [X] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by corybudd-ch** [comment]
2 smelly code smells to fix there highlighted by Sonarqube 👃

File: src/utils/beneficial.owner.statements.ts [hunk 1]
```diff
@@ -6,33 +6,40 @@ import { logger } from "../utils/logger";
...
     if (noChangeBackLink){
-      BACK_LINK = noChangeBackLink;
+      backLinkUrl = noChangeBackLink;
       templateName = config.UPDATE_NO_CHANGE_BENEFICIAL_OWNER_STATEMENTS_PAGE;
       noChangeFlag = true;
     } else {
-      BACK_LINK = registrationFlag ? config.ENTITY_URL : config.OVERSEAS_ENTITY_REVIEW_URL;
+      if (isActiveFeature(config.FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION)){
+        backLinkUrl = containsTrustData(getTrustArray(appData))
+          ? config.UPDATE_TRUSTS_ASSOCIATED_WITH_THE_OVERSEAS_ENTITY_URL
```
Comment (corybudd-ch) [inline src/utils/beneficial.owner.statements.ts:29]: should there be a trusts flag check here as well 🤢

File: src/controllers/update/beneficial.owner.type.controller.ts [hunk 2]
```diff
@@ -67,7 +67,9 @@ export const postSubmit = (req: Request, res: Response) => {
   logger.debugRequest(req, `${req.method} ${req.route.path}`);
   const appData: ApplicationData = getApplicationData(req.session);
   const requiresTrusts: boolean = checkEntityRequiresTrusts(appData);
-  let redirectUrl = config.UPDATE_CHECK_YOUR_ANSWERS_URL;
+  let redirectUrl = (isActiveFeature(config.FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION)
```
Comment (AndreKramer7654) [inline src/controllers/update/beneficial.owner.type.controller.ts:70]: don't need outer ()

File: src/utils/beneficial.owner.statements.ts (hunk 1 above)
Comment (AndreKramer7654) [reply src/utils/beneficial.owner.statements.ts:29]: or can we assume that if update trusts feature flags is off we'll never have trust data?

Comment (joshuahigginsonCH) [reply src/utils/beneficial.owner.statements.ts]: Not really as if they dont have any bo's with trust NOC's then they would've never went through the trust flow, so no trusts @AndreKramer7654

File: src/utils/beneficial.owner.statements.ts [hunk 3]
```diff
@@ -6,33 +6,42 @@ import { logger } from "../utils/logger";
...
-
+    const appData = getApplicationData(req.session);
     if (noChangeBackLink){
-      BACK_LINK = noChangeBackLink;
+      backLinkUrl = noChangeBackLink;
       templateName = config.UPDATE_NO_CHANGE_BENEFICIAL_OWNER_STATEMENTS_PAGE;
       noChangeFlag = true;
     } else {
-      BACK_LINK = registrationFlag ? config.ENTITY_URL : config.OVERSEAS_ENTITY_REVIEW_URL;
+      if (isActiveFeature(config.FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION)){
```
Comment (AndreKramer7654) [inline src/utils/beneficial.owner.statements.ts:27]: add space )){ or remove space on next line

Comment (joshuahigginsonCH) [reply src/utils/beneficial.owner.statements.ts]: fixed

**Review by AndreKramer7654 — APPROVED** [review]

**Review by cdrainCH — APPROVED** [review]

**Review by stephanie-ch — APPROVED** [review]

**Pull request merged**